Remove track tab and rounded corners

pierricgimmig opened the pull request:
This PR's goal is to gain vertical real-estate by removing track tabs in
favor of a track header that is now aligned with tracks, to their left.
This is also similar to how most track UI applications display tracks. 

Note that there are many things we could do to enhance the look of 
tracks, like fine tuning text and collapsible arrow position, coloring on
hover, better separation of header and track. All those changes are 
welcome but outside the scope of this PR. This basically takes our existing
tabs and moves them to the left of the tracks.

![image](https://user-images.githubusercontent.com/3687222/207090058-536ffe71-f074-49ea-95e7-938cc7fe2342.png)

beckerhe commented on src/OrbitGl/CaptureWindow.cpp:
```diff
@@ -725,8 +727,11 @@ void CaptureWindow::RenderSelectionOverlay() {
   uint64_t max_time = std::max(select_start_time_, select_stop_time_);
 
   float from_world = time_graph_->GetWorldFromTick(min_time);
+  from_world = time_graph_->ClampToTimelineUiElementWorldX(from_world);
   float to_world = time_graph_->GetWorldFromTick(max_time);
+  to_world = time_graph_->ClampToTimelineUiElementWorldX(to_world);
   float stop_pos_world = time_graph_->GetWorldFromTick(select_stop_time_);
+  stop_pos_world = time_graph_->ClampToTimelineUiElementWorldX(stop_pos_world);
```
I guess the old calls can now be removed?

florian-kuebler replied:
well those are still used here, right?

beckerhe replied:
Ohhh, I get it now. Sorry never mind.

You can consider writing it in a single expression (or not - what you deem is best).

beckerhe reviewed (approved):
Awesome. This is a cool change. I like the new layout. Code-wise this also looks good to me, but be aware that I don't know a lot about the TimeGraph code. It would be great if you could wait for a review from Florian or Pablo as well. I added one comment, otherwise LGTM.

One more thing: The labels in the timers in your screenshot seem to start before the actual timer box. Is that an unrelated bug or could this be triggered by your margin changes?

florian-kuebler commented on src/OrbitGl/Track.cpp:
```diff
@@ -57,57 +57,39 @@ void Track::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer& text_r
 
-  header_->SetWidth(layout_->GetTrackTabWidth());
-  header_->SetPos(GetPos()[0] + layout_->GetTrackTabOffset(), GetPos()[1]);
+  header_->SetWidth(layout_->GetLeftMargin());
```
Again, I am a bit confused that the layout's margin is used for padding inside the track.

pierricgimmig replied:
changed to TrackHeaderWidth()

reichlfl reviewed (commented):
Hey Pierric, thanks for the change! Looks good, I have only minor comments. But before I approve, could you please check two things:

1) I can't judge how this looks for sub-tracks, could you provide a screenshot?
2) Does the accessibility for all elements still work correctly? Unfortunately this can only be tested on Windows afaik (using e.g. the Accessibility Insights tool and verifying that all the bounding boxes of the tracks and headers are correct).

As for 2, if this is not the case (or problematic to test) we need to discuss if we still need this as it has only been used for E2E test automation.

pierricgimmig commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -897,10 +903,19 @@ void TimeGraph::DrawMarginsBetweenChildren(
   // Left margin mask for Timeline.
   float left_margin_width = layout_->GetLeftMargin();
-  float left_margin_height = GetHeight();
+  float left_margin_height = timeline_ui_->GetHeight();
```
This is for the small portion at the left of the timeline and slider. It is possible for tracks to draw in those regions when scrolling vertically. Previously that was hidden by a single rectangle; now we need two because the tracks don't need to have a second masking rectangle to their left, they already have the header, which also leaves space between tracks unmasked, which is nice to have.

pierricgimmig commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -897,10 +903,19 @@ void TimeGraph::DrawMarginsBetweenChildren(
   Quad left_margin_box = MakeBox(left_margin_pos, Vec2(left_margin_width, left_margin_height));
   primitive_assembler.AddBox(left_margin_box, GlCanvas::kZValueMargin, GlCanvas::kBackgroundColor);
+
+  // Left margin mask for horizontal slider.
```
Not the slider, but track content can spill to that area when scrolling vertically, see comment above.

pierricgimmig commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -119,13 +88,10 @@ void orbit_gl::TrackHeader::DoUpdateLayout() {
   const float button_offset = layout_->GetCollapseButtonOffset();
   const float size = layout_->GetCollapseButtonSize(track_->GetIndentationLevel());
-  const float toggle_y_pos = GetPos()[1] + half_label_height - size / 2;
+  const float toggle_y_pos = GetPos()[1] + layout_->GetTextBoxHeight() * 0.5f;
```
will look into it, yep, thanks.

pierricgimmig commented on src/OrbitGl/include/OrbitGl/TrackContainer.h:
```diff
@@ -112,6 +113,7 @@ class TrackContainer final : public CaptureViewElement {
 
   float vertical_scrolling_offset_ = 0;
   float height_ = 0;
+  float header_width_ = 100.f;
```
it's not, removed, thanks

pierricgimmig commented:
> Does the accessibility for all elements still work correctly? 

@reichlfl , it looks like it:

<img width="1920" alt="header_access" src="https://user-images.githubusercontent.com/3687222/207375299-3213248b-4025-4998-9941-45a7e1eb55a9.png">
<img width="1918" alt="track_access" src="https://user-images.githubusercontent.com/3687222/207375320-f416860b-6145-443d-9e2e-fb9ad8cebab9.png">
<img width="675" alt="access_test" src="https://user-images.githubusercontent.com/3687222/207375331-c88d4a66-2fea-438c-8971-a39f6d3a807a.png">

pierricgimmig commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -726,10 +732,10 @@ void TimeGraph::UpdateChildrenPosAndContainerSize() {
-  track_container_->SetPos(timegraph_current_x + x_offset, timegraph_current_y);
+  track_container_->SetWidth(GetWidth() - total_right_margin);
+  track_container_->SetPos(timegraph_current_x, timegraph_current_y);
+  track_container_->SetHeaderWidth(layout_->GetLeftMargin());
```
changed to TrackHeaderWidth()

reichlfl commented:
> > Does the accessibility for all elements still work correctly?
> 
> @reichlfl , it looks like it:

Cool, thanks for checking! Can you check the `TimerPane` as well? That should be the area which only contains the timers, it might now span the whole track. That would probably be an easy fix within the `AccessibleTrack` implementation - if not, I also think it's not a big deal because the tests (AFAIK) only test for existence, not exact extents of this.

pierricgimmig commented:
> I can't judge how this looks for sub-tracks, could you provide a screenshot?

@reichlfl , @florian-kuebler , how can I test this? Memory track?

pierricgimmig commented:
> One more thing: The labels in the timers in your screenshot seem to start before the actual timer box. Is that an unrelated bug or could this be triggered by your margin changes?

@beckerhe , I noticed that too, but this is unrelated to this change. I think this comes from the Qt text rendering change, I've noticed other artifacts like text overlapping sometimes.

pierricgimmig commented on src/OrbitGl/include/OrbitGl/TrackHeader.h:
```diff
@@ -36,7 +36,7 @@ class TrackHeader : public CaptureViewElement, public std::enable_shared_from_th
   [[nodiscard]] TriangleToggle* GetCollapseToggle() { return collapse_toggle_.get(); }
 
-  [[nodiscard]] float GetHeight() const override { return layout_->GetTrackTabHeight(); }
+  [[nodiscard]] float GetHeight() const override { return GetParent()->GetHeight(); }
```
Seems to work, see screenshots below.

reichlfl commented:
> 

Memory track is probably the easiest, yes

pierricgimmig commented:
> 

@reichlfl , the memory track seems a bit broken, thanks for pointing this out, will fix soon. Cheers!

pierricgimmig commented:
The memory track issue was the legend being drawn outside the track, it is fixed now:

file:///home/pierric/Pictures/Screenshots/mem_track.png

I still have to check for track that contain children, not sure how to do that easily, I'll try setting up a hierarchy in code.

pierricgimmig commented:
@reichlfl , @florian-kuebler, I don't see how I can test subtracks on a generic Linux capture, would you know how? Would you have some test data perhaps, or maybe you could compile this branch and test on your side? Thanks.

florian-kuebler commented:
https://github.com/google/orbit/blob/main/contrib/automation_tests/testdata/OrbitTest_1-72.orbit could be a good example.

But, Ill also check the gpu tracks tomorrow.

florian-kuebler reviewed (changes requested):
Actually there are some issues with subtracks left.

1. On the OrbitTest_1-72 capture the page fault track shows some issues. Subtrack heads is on the tracks header and also the legend is drawn on top of the data. See screenshot:
![Screenshot 2022-12-15 at 16 25 03](https://user-images.githubusercontent.com/2725914/207901012-1ac7094a-aea4-4424-a4f1-cac9b79becfb.png)

2. There is a crash for captures with the gpu (subtrack). I will look into this, briefly, but fear that I don't have a shareable capture at hand. (I will try to get something though).
![Screenshot 2022-12-15 at 16 33 06](https://user-images.githubusercontent.com/2725914/207901680-910a2cc2-efd6-4bef-9728-a699fe748f87.png)

In general, I like that we are trying to get more vertical space. But also I feel that the space on the left now is pretty big, considering we still have the split between timeline and the other data views.
Couldn't we do the middle ground: Have the bare minimum needed for the track headers. Like one or two pixels between tracks; remove the rounded corners for the header and also make the headers smaller?

pierricgimmig commented:
Thanks for checking @reichlfl . I'll try out the data and fix the issues. Concerning the big headers, I also found the same at first, but got really accustomed to it using other tools that have this layout. I think it might be a great place to put extra information about track stats too. One thing I want to add is the ability to resize the header part, so we could go from what is suggested to what you are suggesting when minimizing the header width.

florian-kuebler commented:
>One thing I want to add is the ability to resize the header part, so we could go from what is suggested to what you are suggesting when minimizing the header width.

But then why scraping out the current header, when we are going to add it back again? Maybe we should collect a bunch of screenshots from other tools with this track like format, and see if we find a common sense (as said, I am a big fan of reducing vertical gaps). I looked at tools like final cut pro (which has the header) and fruity loops (which has the left part, but much smaller).

florian-kuebler commented on src/OrbitGl/include/OrbitGl/TrackHeader.h:
```diff
@@ -36,7 +36,7 @@ class TrackHeader : public CaptureViewElement, public std::enable_shared_from_th
   [[nodiscard]] TriangleToggle* GetCollapseToggle() { return collapse_toggle_.get(); }
 
-  [[nodiscard]] float GetHeight() const override { return layout_->GetTrackTabHeight(); }
+  [[nodiscard]] float GetHeight() const override { return GetParent()->GetHeight(); }
```
That one here is actually the issue that leads to the crash.

Basically `GpuSubmissionTrack::GetHeight` is calling `header_->GetHeight()`, which again with this line here is calling `GpuSubmissionTrack::GetHeight` (as the track is the parent of the header.

florian-kuebler replied:
@reichlfl FYI

florian-kuebler replied:
Adding a quick fix (which should not be the way to go), we see the same issues as in the memory track:
![Screenshot 2022-12-16 at 11 58 03](https://user-images.githubusercontent.com/2725914/208083810-1d1aef38-ba91-4010-b104-bd0a5e17c19e.png)

florian-kuebler replied:
@pierricgimmig This crash here does not seem to be fixed.

pierricgimmig replied:
Ah ok, hopefully I can repro with @ronaldfw 's capture.

florian-kuebler replied:
So, I looked into your fix for the crash. I still think this is against the layouting principles we came up with. The idea about our parent/child relationship is, that the position and size is controlled by the parent (this is mostly due to the lack of flexible layouting), and we tried to be consistent about this.

Now with your change you must remember that you can't call `GetHeight` for you the header, while it is fine and maybe even required to call the same method for your other childs (like for subtracks).

Maybe @reichlfl has a better idea how to implement the `TrackHeader::GetHeight` method.

florian-kuebler replied:
I gave this another thought. The track header is completely inside the track (which is correct). It's relevant height however, is not the complete track's height, but only the height of the label (or the toggle) + some slight spacing at the top and below the label.

So you can implement this method with information locally known here.

This also allows to fix the issues with the header being out of the track in case of subtracks, as we can move the subtracks below the parent's header when the parent track is expanded.

florian-kuebler replied:
I guess the easy fix would be to explicitly set the header's height in `Track::DoUpdateLayout()`.

Alternatively, @reichlfl mentioned that we could do something similar to the capture view element's `width` using the `LayoutFlags`.

But, I guess the simple solution would work for now.

reichlfl replied:
@pierricgimmig I'm not sure if this discussion happened before or after you requested the last review - just to be sure this doesn't get lost, as far as I see it this is the main conceptual issue that prevents this from being merged. Children cannot use the parent height to determine their own height, only the other way around. Please instead introduce a `SetHeight` method for the headers, and call this during `Track::DoUpdateLayout`. Make sure to set the "update required" flags accordingly within `SetHeight` (see other layout-changing setters for reference).

pierricgimmig replied:
Implemented your suggestion regarding a SetHeight method @reichlfl

florian-kuebler commented:
Just for reference. This is how a more condensed ui with the current implementation logic could look like:
![Screenshot 2022-12-16 at 12 51 33](https://user-images.githubusercontent.com/2725914/208092492-b430d070-352a-4786-8c87-8d9823100e63.png)

I made all the gaps smaller, removed the rounding, reduced the size of the triangle and shifted the header to the left (including reducing the width).

florian-kuebler commented:
Also, I extracted the property controls from here to https://github.com/google/orbit/pull/4600, as it is nice for testing/playing around.

pierricgimmig commented:
@florian-kuebler , thanks looking into this, I'll fix the remaining issues with the left headers. And then we can compare the different  approaches.

pierricgimmig commented:
@florian-kuebler , I thought about this a bit more, I think having the track headers to the left is really the way to go. It makes everything clearer, there is a room for additional info and icons (a pin for example, or a visibility toggle), and it's just a cleaner layout. I don't think horizontal real estate is as important as vertical real estate. I thought it was when I originally implemented the track tabs, but it's not. I think Perfetto is a really good example at a clean layout that we should move towards. There are many other examples of track UI that use the same layout. I'll continue fixing the remaining bugs in the current PR. Please let me know if any internal discussion oppose the current change. Thanks.

pierricgimmig commented:
@reichlfl , @florian-kuebler , the PR is in a better state, I'd love to get your feedback. Thanks!

florian-kuebler commented:
> @reichlfl , @florian-kuebler , the PR is in a better state, I'd love to get your feedback. Thanks!

Thanks. I'll have a look and test it tomorrow.

pierricgimmig commented:
Thanks for the capture @ronaldfw!

florian-kuebler commented:
Thanks for the changes so far. I played around with the current state, and mostly like it.

Despite my comment on the `TrackHeader::GetHeight` method, which I think must be resolved, I found three remaining bugs:
1. If you expand the GpuTrack and then collapse the submission subtrack, the header position is off (see screenshot). The same happens, if the submissions only have a depth of 1.
![Screenshot 2022-12-21 at 08 13 13](https://user-images.githubusercontent.com/2725914/208850863-8d3bab22-c6b3-4a02-9fab-7cf7beb62518.png)
1. Similarly, if you have a track with gpu debug markers and have that one collapsed, the position of the header is also off:
 ![Screenshot 2022-12-21 at 08 21 30](https://user-images.githubusercontent.com/2725914/208851331-d49d7f51-f0bd-4781-bad3-be63566e9600.png)
1. If you have empty thread tracks, that only have thread state, there is a spacing at the bottom missing:
![Screenshot 2022-12-21 at 08 17 57](https://user-images.githubusercontent.com/2725914/208851527-5410f91d-63f0-4fac-8a76-f9d36f285cc4.png)

Also, I do like the new indicators for expand/collapse. I am wondering whether they should be swapped, however. @reichlfl what are your thoughts on that?

Finally, I would decrease the width of the header from 350 to 250.

florian-kuebler commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -42,53 +42,24 @@ void TrackHeader::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer&
   // Draw tab background
-  const float label_height = GetHeight();
-  const float half_label_height = 0.5f * label_height;
+  const float label_height = GetParent()->GetHeight();
```
Also here should probably not use the parent's height.

pierricgimmig replied:
done.

florian-kuebler commented on src/OrbitGl/GpuTrack.cpp:
```diff
@@ -95,7 +95,7 @@ void GpuTrack::UpdatePositionOfSubtracks() {
   submission_track_->SetIndentationLevel(indentation_level_ + 1);
 
-  float current_y = pos[1] + layout_->GetTrackTabHeight();
+  float current_y = pos[1];
```
I guess these two changes here result in the problem we've seen with the subtrack's labels being off (or better with the tracks not aligned to the headers).

I guess we want to add a `header_->GetHeight()` here and below, which requires a different implementation of `TrackHeader::GetHeight`.

pierricgimmig commented:
Hi @florian-kuebler , thanks for your comments. I fixed 1, 2 and 3. I didn't want to spend too much time on 2 as I think the subtracks should be completely overhauled in favor of a full track hierarchy where subtracks are treated as full-fledged tracks, with their own header. Thanks again for your reviews.

reichlfl commented:
> I am wondering whether they should be swapped, however

Apparently the current version is also what the latest Material UI accordion elements use. Even though I find this confusing, I guess we should stick with it. To be sure: "arrow down" means it's currently collapsed, "arrow up" means it's expanded, right?

karupayun commented on src/OrbitGl/ThreadTrack.cpp:
```diff
@@ -271,20 +270,19 @@ void ThreadTrack::UpdatePositionOfSubtracks() {
+  float current_y = GetPos()[1] + layout_->GetTrackContentTopMargin();
 
-  thread_state_bar_->SetPos(pos[0], current_y);
+  thread_state_bar_->SetPos(pos[0] + header_->GetWidth(), current_y);
```
Since you are always using `pos[0] + header_->GetWidth()` and to avoid repetition of code, I would prefer to include the header_width in the variable pos.

pierricgimmig replied:
done.

karupayun commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -877,7 +882,9 @@ void TimeGraph::DoDraw(orbit_gl::PrimitiveAssembler& primitive_assembler,
   // being drawn partially outside the TrackContainer space. This hack is needed until we assure
   // that no element is drawn outside of its parent's area.
-  DrawMarginsBetweenChildren(primitive_assembler);
+  if (layout_->GetDrawTimeGraphMasks()) {
```
Why we need this?

pierricgimmig replied:
For debugging

pierricgimmig replied:
What I mean is that it's a useful debugging feature, the code is meant to be merged.

karupayun commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -42,53 +42,24 @@ void TrackHeader::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer&
-    return;
+  if (track_->GetIndentationLevel() == 0 && layout_->GetDrawTrackHeaderBackground()) {
+    Quad box = MakeBox(Vec2(indentation_x0, y0), Vec2(label_width, label_height));
+    primitive_assembler.AddBox(box, track_z, track_->GetTrackBackgroundColor(), shared_from_this());
```
What do you think about choosing a slightly different color for the TrackTab to differentiate it with its content?

pierricgimmig replied:
Yeah that would be cool, would have been easy to set colors with ImGui, but ImGui is not there anymore. Let's do this in an other PR.

karupayun commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -119,25 +91,34 @@ void orbit_gl::TrackHeader::DoUpdateLayout() {
   collapse_toggle_->SetWidth(size);
   collapse_toggle_->SetHeight(size);
-  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1]);
+  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1] + GetVerticalLabelOffset());
```
It would be better to include this offset in the calculation of the position (`toggle_y_pos`)

pierricgimmig replied:
Done.

karupayun commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -119,25 +91,34 @@ void orbit_gl::TrackHeader::DoUpdateLayout() {
-  const float toggle_y_pos = GetPos()[1] + half_label_height - size / 2;
-  Vec2 toggle_pos = Vec2(x0 + button_offset, toggle_y_pos);
+
+  constexpr float kOffsetX = -0.5f;
```
Could you explain why this offsets with a comment?

pierricgimmig replied:
this was an attempt to fix the asymmetry, but it doesn't work all the time, so I'm removing it.

karupayun commented on src/OrbitGl/TrackHeader.cpp:
```diff
@@ -119,25 +91,34 @@ void orbit_gl::TrackHeader::DoUpdateLayout() {
   collapse_toggle_->SetIsCollapsible(track_->IsCollapsible());
 }
 
+float TrackHeader::GetVerticalLabelOffset() const {
```
This only works when IndentationLevel is 1. Please solve the issue or include that information in the comment and log an error in case that IndentationLevel is higher than 1.

pierricgimmig replied:
Added a log as fixing this properly is a change that will come in its own PR, basically overhauling the track hierarchy system.

karupayun commented on src/OrbitQt/TimeGraphLayoutWidget.cpp:
```diff
@@ -32,14 +32,26 @@ TimeGraphLayoutWidget::TimeGraphLayoutWidget(QWidget* parent)
   AddWidgetForProperty(&rounding_radius_);
   AddWidgetForProperty(&rounding_num_sides_);
   AddWidgetForProperty(&text_offset_);
-  AddWidgetForProperty(&left_margin_);
```
You should also erase the widget for track_tab_width_, track_tab_height_ and track_tab_offset_, or I am missing something?

pierricgimmig replied:
good point, done, thanks.

karupayun reviewed (commented):
Nice PR Pierric! At first I was unsure of the new layout, but I think that I'm ok with it now. I would include a different color for the header and the content, so it's easier to understand the border. What do you think?

A few more details:
1-
![7YcUWgzarMKeBHX](https://user-images.githubusercontent.com/8610429/210280144-69aa8580-8e94-4c3d-89d8-21e3f4ffcfa9.png)

The green line with the mouse can be seen a bit more to the left than the content, do you know why is that?

2-
![5HuS5jW7fr4v3Ko](https://user-images.githubusercontent.com/8610429/210280172-5873649f-e8b0-4f00-a2aa-38fc31959a81.png)

The toggles of the subtracks seems a bit broken.


3-
![E6nDRVz8PPNoom9](https://user-images.githubusercontent.com/8610429/210280195-1e03d176-5a01-43ea-b379-2442164c2ece.png)

The title of the FrameTrack intersect with the average time in the most common case, could you move this line to be only drawn in the content part?

4-
![5FNmxP4saaDNM92](https://user-images.githubusercontent.com/8610429/210280251-3326efdb-4e6b-4164-aa9f-93bb969fa1cb.png)

There are some broken cases with the Gpu-subtracks. Here it's an example with the track expanded but both subtracks contracted.

pierricgimmig commented:
@reichlfl , @karupayun, thanks for the review and comments, I'll address them soon, I might revert the arrows for now too. As pointed out, they seem to render weirdly sometimes. I don't fully understand why, but it seems primitives are sometimes off by one pixel.

pierricgimmig commented on src/OrbitGl/GraphTrack.cpp:
```diff
@@ -231,9 +231,8 @@ void GraphTrack<Dimension>::DrawLegend(PrimitiveAssembler& primitive_assembler,
-  float x0 = GetPos()[0] + layout_->GetRightMargin();
-  const float y0 =
-      header_->GetPos()[1] + header_->GetHeight() + layout_->GetTrackContentTopMargin();
+  float x0 = GetPos()[0] + +header_->GetWidth() + layout_->GetRightMargin();
```
done

pierricgimmig commented on src/OrbitGl/CaptureWindow.cpp:
```diff
@@ -493,6 +495,13 @@ void CaptureWindow::Draw(QPainter* painter) {
   uint64_t start_time_ns = orbit_base::CaptureTimestampNs();
   bool time_graph_was_redrawn = time_graph_ != nullptr && time_graph_->IsRedrawNeeded();
 
+  if (time_graph_layout_->GetDrawAsIfPicking()) {
```
simplified the code

pierricgimmig commented:
@reichlfl , @karupayun, @florian-kuebler , I addressed most of the comments. Concerning the GPU subtracks issue, is that blocking? I feel like it might be faster if someone who knows the code better implemented a quick fix rather than going back and forth...  Also, I rebased this change in another PR if you want to see it with the latest UI changes. thanks for your help!

pierricgimmig commented:
> The green line with the mouse can be seen a bit more to the left than the content, do you know why is that?

N/A in the rebase branch, and I couldn't repro on Windows weirdly.

pierricgimmig commented:
> The toggles of the subtracks seems a bit broken.

Reverted, the ceiling code was the reason behind the weirdness.

pierricgimmig commented:
> The title of the FrameTrack intersect with the average time in the most common case, could you move this line to be only drawn in the content part?

fixed

florian-kuebler commented:
Thanks for addressing the comments. I think it looks good now, but I want to do another full review. Shall I do it here, or at the #4727 pr?

pierricgimmig commented:
Closing as https://github.com/google/orbit/pull/4727 supersedes this PR.
